pyphare/pyphare/pharein/simulation.py

```diff
@@ -314,24 +369,36 @@ def check_refinement(**kwargs):
 
 
 
+def check_optional_keywords(**kwargs):
+    extra = []
+
+    if check_refinement(**kwargs) != "boxes":
+        extra += ['max_nbr_levels']
+
+    return extra
+
 
 # ------------------------------------------------------------------------------
 
 def checker(func):
     def wrapper(simulation_object, **kwargs):
         accepted_keywords = ['domain_size', 'cells', 'dl', 'particle_pusher', 'final_time',
                              'time_step', 'time_step_nbr', 'layout', 'interp_order', 'origin',
-                             'boundary_types', 'refined_particle_nbr', 'path',
-                             'diag_export_format', 'max_nbr_levels', 'refinement_boxes','refinement',
+                             'boundary_types', 'refined_particle_nbr', 'path', 'nesting_buffer',
+                             'diag_export_format', 'refinement_boxes', 'refinement',
                              'smallest_patch_size', 'largest_patch_size', "diag_options" ]
 
+        accepted_keywords += check_optional_keywords(**kwargs)
+
         wrong_kwds = phare_utilities.not_in_keywords_list(accepted_keywords, **kwargs)
         if len(wrong_kwds) > 0:
             raise ValueError("Error: invalid arguments - " + " ".join(wrong_kwds))
 
         dl, cells = check_domain(**kwargs)
         kwargs["dl"] = dl
         kwargs["cells"] =  cells
+        kwargs["refinement_ratio"] = 2
+        kwargs["nesting_buffer"] = kwargs.get('nesting_buffer', 0)
```

Comment:
SAMRAI doc says nesting_buffer is one int per level :/

Reply:
updated to putIntVector in C++ hierarchy